jest/jest.d.ts

```diff
@@ -1,126 +1,88 @@
+        toMatchSnapshot(): boolean;
+        toThrow(): boolean;
+        toThrowError(expected: string | RegExp): boolean;
+        toThrowError<TFunction extends Function>(expected: TFunction): boolean;
     }
-
-    interface It {
-        (name: string, fn: EmptyFunction): void;
-        only(name: string, fn: EmptyFunction): void;
+    
+    interface MockContext<T> {
+        calls: any[][];
+        instances: T[];
     }
 
     interface Mock<T> {
         new (): T;
-        (...args: any[]): any; // TODO please fix this line! added for TypeScript 1.1.0-1 https://github.com/DefinitelyTyped/DefinitelyTyped/pull/2932
+        (...args: any[]): any; // Making Mock Callable and fixing: Value of type 'Mock<T>' is not callable.
         mock: MockContext<T>;
         mockClear(): void;
         mockImplementation(fn: Function): Mock<T>;
-        mockImpl(fn: Function): Mock<T>;
+        mockImplementationOnce(fn: Function): Mock<T>;
         mockReturnThis(): Mock<T>;
         mockReturnValue(value: any): Mock<T>;
         mockReturnValueOnce(value: any): Mock<T>;
     }
-
-    interface MockContext<T> {
-        calls: any[][];
-        instances: T[];
-    }
-
-    // taken from Jasmine since addMatchers calls into the jasmine api
-    interface CustomMatcherFactories {
-        [index: string]: CustomMatcherFactory;
-    }
-
-    // taken from Jasmine since addMatchers calls into the jasmine api
-    interface CustomMatcherFactory {
-        (util: MatchersUtil, customEqualityTesters: Array<CustomEqualityTester>): CustomMatcher;
-    }
-
-    // taken from Jasmine since addMatchers calls into the jasmine api
-    interface MatchersUtil {
-        equals(a: any, b: any, customTesters?: Array<CustomEqualityTester>): boolean;
-        contains<T>(haystack: ArrayLike<T> | string, needle: any, customTesters?: Array<CustomEqualityTester>): boolean;
-        buildFailureMessage(matcherName: string, isNot: boolean, actual: any, ...expected: Array<any>): string;
-    }
-
-    // taken from Jasmine since addMatchers calls into the jasmine api
-    interface CustomEqualityTester {
-        (first: any, second: any): boolean;
-    }
-
-    // taken from Jasmine since addMatchers calls into the jasmine api
-    interface CustomMatcher {
-        compare<T>(actual: T, expected: T): CustomMatcherResult;
-        compare(actual: any, expected: any): CustomMatcherResult;
-    }
-
-    // taken from Jasmine since addMatchers calls into the jasmine api
-    interface CustomMatcherResult {
-        pass: boolean;
-        message: string;
-    }
+    
+    function clearAllTimers(): void;
+    function disableAutomock(): void;
+    function enableAutomock(): void;
+    function fn<T>(implementation?: Function): Mock<T>;
+    function isMockFunction(fn: Function): boolean;
+    function genMockFromModule<T>(moduleName: string): T;
+    function mock(moduleName: string, factory?: Function, options?: {virtual: boolean}): void;
```

Comment:
I've searched the whole API Reference and haven't found a single example showing or mentioning possible chaining. So I guest this is undocomented behaviour ... If you can provide any documentation rather than actual usage I think it's better off to `return void`.